windtools/amrwind/post_processing.py

```diff
@@ -29,26 +46,63 @@ def _setup_time_coords(self,ds):
         return ds
 
-    def _load_timeseries(self):
-        ds = xr.load_dataset(self.fpath)
-        ds = self._setup_time_coords(ds)
-        self.ds = ds
-
-    def _load_timeheight_profiles(self):
-        ds = xr.load_dataset(self.fpath, group='mean_profiles')
-        ds = ds.rename({'h':'height'})
-        times = self.ds.coords[self.time_coord].values
-        ds = ds.assign_coords({
-            self.time_coord: ('num_time_steps',times),
-            'height': ds['height'],
-        })
-        ds = ds.swap_dims({'num_time_steps':self.time_coord, 'nlevels':'height'})
-        ds = ds.transpose(self.time_coord,'height')
-        self.ds = xr.combine_by_coords([self.ds, ds])
-        self.ds[self.time_coord] = np.round(self.ds[self.time_coord],5)
-
-
-
+    def _load_timeseries(self,load_mean_profiles=False):
+        dslist = []
+        for fpath in self.fpaths:
+
+            ds = xr.load_dataset(fpath)
+            ds = self._setup_time_coords(ds)
+
+            if load_mean_profiles:
+                pro = xr.load_dataset(fpath, group='mean_profiles')
+                pro = pro.rename({'h':'height'})
+                pro = pro.assign_coords({
+                    self.time_coord: ('num_time_steps',
+                                      ds.coords[self.time_coord].values),
+                    'height': pro['height'],
+                })
+                pro = pro.swap_dims({'num_time_steps':self.time_coord,
+                                     'nlevels':'height'})
+                # make sure underlying array data have the expected shape
+                pro = pro.transpose(self.time_coord,'height')
+                # merge time-height profiles with timeseries
+                ds = xr.combine_by_coords([ds, pro])
+
+            dslist.append(ds)
+
+        if len(dslist) > 1:
+            for ds,fpath in zip(dslist,self.fpaths):
+                # conflicting attrs causes a merge error
+                print('Merging', fpath, ds.attrs.pop('created_on').rstrip())
+            # concat and merge
+            self.ds = xr.combine_by_coords(dslist)
+        else:
+            self.ds = dslist[0]
+
+    def _calc_total_fluxes(self):
+        for varn in self.ds.data_vars:
+            if varn.endswith('_r'):
+                varn_sfs = varn[:-2] + '_sfs'
+                if varn_sfs in self.ds.data_vars:
+                    varn_tot = varn[:-2] + '_tot'
+                    self.ds[varn_tot] = self.ds[varn] + self.ds[varn_sfs]
+
+    def _calc_TI(self):
+        ang = np.arctan2(self.ds['v'], self.ds['u'])
+        rotatedvar = self.ds["u'u'_r"] * np.cos(ang)**2 \
+                   + self.ds["u'v'_r"] * 2*np.sin(ang)*np.cos(ang) \
+                   + self.ds["v'v'_r"] * np.sin(ang)**2
+        self.ds['TI'] = np.sqrt(rotatedvar) / self.ds['hvelmag']
+
+    def __getitem__(self,key):
+        return self.ds[key]
+
+    def rolling_mean(self,Tavg=3600.):
+        dt = float(self.t[1] - self.t[0])
```

Comment:
You are assuming the time step is fixed. This is usually not the case with spinning up precursors.

Reply:
I've generalized this now...